doc/user_guide/model.rst

```diff
@@ -659,65 +653,88 @@ Fitting the model to the data
-    | "L-BFGS-B"               |  Yes   | No               | 'ls', 'ml', 'custom' | local  |
-    +--------------------------+--------+------------------+----------------------+--------+
-    | "TNC"                    |  Yes   | No               | 'ls', 'ml', 'custom' | local  |
-    +--------------------------+--------+------------------+----------------------+--------+
-    | "Differential Evolution" |  Yes   | No               | 'ls', 'ml', 'custom' | global |
-    +--------------------------+--------+------------------+----------------------+--------+
-
+.. table:: Features of supported curve-fitting optimizers.
+
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | Optimizer                            | Bounds | Gradients | Errors | Loss function  | Type   |
+    +======================================+========+===========+========+================+========+
+    | ``"lm"`` (default)                   |  Yes   | Yes       | Yes    | Only ``"ls"``  | local  |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | ``"trf"``                            |  Yes   | Yes       | Yes    | Only ``"ls"``  | local  |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | ``"dogbox"``                         |  Yes   | Yes       | Yes    | Only ``"ls"``  | local  |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | ``"odr"``                            |  No    | Yes       | Yes    | Only ``"ls"``  | local  |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | :py:func:`scipy.optimize.minimize`   |  Yes * | Yes *     | No     | All            | local  |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | ``"Differential Evolution"``         |  Yes   | No        | No     | All            | global |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | ``"Dual Annealing"`` **              |  Yes   | No        | No     | All            | global |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+    | ``"SHGO"`` **                        |  Yes   | No        | No     | All            | global |
+    +--------------------------------------+--------+-----------+--------+----------------+--------+
+
+.. note::
+
+    \* **All** of the fitting algorithms available in :py:func:`scipy.optimize.minimize` are currently
+    supported by HyperSpy; however, only some of them support bounds and/or gradients. For more information,
+    please see the `SciPy documentation <http://docs.scipy.org/doc/scipy/reference/optimize.html>`_.
+
+    \*\* Requires ``scipy >= 1.2.0``.
```

Comment:
Scipy 1.5.2 is out, and I notice that hyperspy [setup.py currently requires scipy 0.15] (released 2015)(https://github.com/hyperspy/hyperspy/blob/fe7602e4ff9ae51a86c2a4c0d0545971ef972ae3/setup.py#L48). Maybe we can just set scipy 1.2 as required version? I notice conda installs scipy 1.5.0.

Reply:
I have no objection, I was conservative in advancing it to 0.17.0 here. If @francisco-dlp and @ericpre have no objections/reasons why this is a bad idea, then we can advance it further

Reply:
Ubuntu 18.04 LTS have [scipy 0.19.1](https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=python3-scipy). We are not dealing dependencies requirement consistently but since there is no significant advantages to change it, I would keep it as broad as possible. Also, I suspect that scipy is one our dependencies which is the most difficult to build.

Reply:
I'll keep it at 0.17